docs/user-documentation/soc-and-feeds.md

```diff
@@ -0,0 +1,126 @@
+
+```js
+const owner = '0x8d3766440f0d7b949a5e32995d09619a7f86e632'
+const socReader = bee.makeSOCReader(owner)
+const identity = '0000000000000000000000000000000000000000000000000000000000000000'
+const soc = await socReader.download(identity)
+const data = soc.payload()
+```
+
+### Writing SOCs
+
+When writing an SOC first we need to make a writer object. Because we need to sign the chunk we need to pass in a `signer` object. The `signer` object can be either an Ethereum private key (as a hex string or `Uint8Array` type) or it can be an instance of the `Signer` interface. The `Signer` interface can be used for integration with 3rd party Ethereum wallet applications, because Swarm uses the same format for signing chunks that Ethereum uses for signing transactions.
+
+```ts
+type SyncSigner = (digest: Uint8Array) => Signature
+type AsyncSigner = (digest: Uint8Array) => Promise<Signature>
+
+/**
+ * Interface for implementing Ethereum compatible signing.
+ *
+ * @property sign     The sign function that can be sync or async
+ * @property address  The ethereum address of the signer
+ */
+export type Signer = {
+  sign: SyncSigner | AsyncSigner
+  address: EthAddress
+}
+```
+
+:::warning Your communication privacy may be at risk
+We suggest to use either ephemeral private keys (e.g. randomly generated) or the `Signer` interface when writing to SOC or Feeds. Never use your real Ethereum private keys here (or in any web applications really) directly because you may lose all your Ethers belonging to it.
+:::
+
+Using the writer interface is then similar to using the reader:
+
+```js
+const signer = '0x634fb5a872396d9693e5c9f9d7233cfa93f395c093371017ff44aa9ae6564cdd'
+const socWriter = bee.makeSOCWriter(signer)
+const identity = '0000000000000000000000000000000000000000000000000000000000000000'
+const data = new Uint8Array([1, 2, 3])
+const response = await socWriter.upload(identity, data)
+```
+
+## Feeds
+
+Feeds are an abstraction built on top of SOCs to provide mutable resources on the otherwise immutable data types that Swarm supports.
+
+A feed is defined by its `owner` (see above), a `topic` (which is 32 bytes arbitrary data, usually expected as a hex string or in a `Uint8Array` type) and a `type`. `type` defines how the index of the feed is stored in Swarm and how the lookup works, currently only the `sequence` type is supported.
+
+Publishers are the single owners of feed chunks and are the only ones able to post updates to their feed. Posting an update requires (1) constructing the identifier from the topic and the correct index and (2) signing it concatenated together with the hash of the arbitrary content of the update.
+
+Conversely, users can consume a feed by retrieving the chunk by its address. Retrieving an update requires the consumer to construct the address from the owner’s public key and the identifier. To calculate the identifier they need the topic and the appropriate index. For this they need to know the indexing scheme.
```

Comment:
Could we try and break this down a bit? I am myself not sure I fully understand it, it just sounds like "you need this, but to get it you need this and this and for that you need also this and this".

Reply:
TBH I copied this verbatim from the Book of Swarm :)
As such I get it that the language is somewhat academic, so suggestions to improve are welcome.

Reply:
@significance ^^?

Reply:
yeah i think it makes sense, but indeed academic and a maybe bit highbrow maybe for the docs.

if i were to suggest anything, maybe just that there is a tl;dr basic explanation at the start which simply states feeds are a key-value store / pub-sub system, then allow people to have a deeper dive if that's something they would be interested in. 

in general i have tried to take the approach of keeping the docs as a simplistic user manual and leaving the deep theory for The Book. some has crept in from various PR's but that was the initial plan 😀